internal/credentials/model.go

```diff
@@ -0,0 +1,96 @@
+package credentials
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/razorpay/metro/internal/common"
+	"github.com/razorpay/metro/pkg/utils"
+	"github.com/sethvargo/go-password/password"
+)
+
+const (
+	// Prefix for all credential keys in the registry
+	Prefix = "credentials/"
+
+	// PartSeparator used in constructing the username parts
+	PartSeparator = "__"
+
+	// perf10__656f81 : sample username format
+	usernameFormat = "%v%v%v"
+
+	// CtxKey identified credentials will be populated in ctx with this key
+	CtxKey = "CredentialsCtxKey"
+)
+
+// Model for a credential
+type Model struct {
+	common.BaseModel
+	Username  string `json:"username"`
+	Password  string `json:"password"`
+	ProjectID string `json:"project_id"`
+	// in future, this model can contain some ACL as well
+}
+
+// ICredentials defines getters on a credential
+type ICredentials interface {
+	GetUsername() string
+	GetPassword() string
+	GetProjectID() string
+}
+
+// Key returns the key for storing credentials in the registry
+func (m *Model) Key() string {
+	return m.Prefix() + m.Username
+}
+
+// Prefix returns the Key prefix
+func (m *Model) Prefix() string {
+	return common.GetBasePrefix() + Prefix + m.ProjectID + "/"
+}
+
+// NewCredential returns a new credential model
+func NewCredential(username, password string) ICredentials {
+	m := &Model{}
+	m.Username = username
+	// store Password only after encoding it
+	m.Password = utils.Encode(password)
```

Comment:
or encrypt them using some standard algorithms AES256 or something

Reply:
This will require additional code changes and testing. Tracking here https://github.com/razorpay/metro/issues/138.